File: kotlin/src/main/kotlin/applications/aoc2023/DayOne.kt
package applications.aoc2023

import model.Calibrator
import model.types.QuestionPart
import model.types.QuestionPart.PART_A
import model.types.QuestionPart.PART_B
import utils.TextFileProcessor

class DayOne {
    private val fileLines = TextFileProcessor(2023, 1).getFileLines()

    fun solvePartA() {
        println("Part A: ${getCalibrationTotal(PART_A)}")
    }

    fun solvePartB() {
        println("Part B: ${getCalibrationTotal(PART_B)}")
    }

    private fun getCalibrationTotal(questionPart: QuestionPart): Int {
        var answer = 0
        for (line in fileLines) {
            val calibrator = Calibrator(line, questionPart)

            val calibrationValue = calibrator.getCalibrationValue()

            answer += calibrationValue
        }

        return answer
    }
}


fun main() {
    DayOne().solvePartA()
    DayOne().solvePartB()
}

File: python/src/main/python/application/aoc2023/day_one.py
from src.main.python.model.types.question_part import QuestionPart
from src.main.python.model.calibrator import Calibrator
from src.main.python.utils.text_file_processor import TextFileProcessor


class DayOne:
    __file_lines = TextFileProcessor(2023, 1).get_file_lines()

    def solve_part_a(self) -> None:
        print(f"Part A: {self.__get_calibration_total(QuestionPart.PART_A)}")

    def solve_part_b(self) -> None:
        print(f"Part B: {self.__get_calibration_total(QuestionPart.PART_B)}")

    def __get_calibration_total(self, question_part: QuestionPart) -> int:
        answer: int = 0
        for line in self.__file_lines:
            calibrator: Calibrator = Calibrator(line, question_part)

            calibration_value: int = calibrator.get_calibration_value()

            answer += calibration_value

        return answer


if __name__ == "__main__":
    DayOne().solve_part_a()
    DayOne().solve_part_b()
